apps/designer/app/shared/nano-states/nano-states.ts

```diff
@@ -31,6 +31,22 @@ export const useSetPresetStyles = (presetStyles: PresetStyles) => {
 };
 
 export const stylesContainer = atom<Styles>([]);
+export const stylesIndexStore = computed(stylesContainer, (styles) => {
+  const stylesByInstanceId = new Map<Instance["id"], Styles>();
+  for (const stylesItem of styles) {
+    const { instanceId } = stylesItem;
+    let instanceStyles = stylesByInstanceId.get(instanceId);
+    if (instanceStyles === undefined) {
+      instanceStyles = [];
+      stylesByInstanceId.set(instanceId, instanceStyles);
+    }
+    instanceStyles.push(stylesItem);
+  }
+  return {
+    stylesByInstanceId,
```

Comment:
could be directly returning styleByIndexId, so that styleByIndexIdStore can be there

Reply:
We may need other indexes in the future. And it may be cheaper to reuse the same loop